yodeploy/cmds/yodeploy_server.py

```diff
@@ -29,11 +30,42 @@ def parse_args(flask_app):
     return opts
 
 
+def create_context():
+    if hasattr(ssl, 'create_default_context'):  # Python >= 2.7.9
+        context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+        context.options |= ssl.OP_CIPHER_SERVER_PREFERENCE
+    else:
+        context = LegacySSLContext()
+
+    context.load_cert_chain(
+        certfile=flask_app.config.server.ssl.cert_chain,
+        keyfile=flask_app.config.server.ssl.key)
+    context.set_ciphers(flask_app.config.server.ssl.ciphers)
+    return context
+
+
+class LegacySSLContext(object):
+    """Wrap ssl.wrap_socket in a class with the SSLContext API"""
+
+    _ciphers = None
+
+    def load_cert_chain(self, certfile, keyfile):
+        self._certfile = certfile
+        self._keyfile = keyfile
+
+    def set_ciphers(self, ciphers):
+        self._ciphers = ciphers
+
+    def wrap_socket(self, sock, **kwargs):
+        return ssl.wrap_socket(
+            sock, certfile=self._certfile, keyfile=self._keyfile,
+            ciphers=self._ciphers, **kwargs)
```

Comment:
Looking around at other people's implementations, [such as this one](https://github.com/pallets/werkzeug/blob/2f5268966428fcfdc76d2311864a7c43cf826c93/werkzeug/serving.py#L431-L450), notice you don't have `ssl_version`. I can't find anywhere that is consumed so I hope all is well.

Reply:
Yes. So this is a clone of werkzeug's (which we were using before). I didn't bother to add ssl_version because the default is `PROTOCOL_SSLv23` (now called `PROTOCOL_TLS` because the old name was confusing). This is a perfectly good default, and I don't want to get caught up in the deprecation of the old name.